python/objToJSON.c

```diff
@@ -579,7 +592,7 @@ void Object_beginTypeContext (JSOBJ _obj, JSONTypeContext *tc)
     return;
   }
   else
-  if (PyString_Check(obj))
+  if (PyString_Check(obj) && !PyObject_HasAttrString(obj, "__json__"))
```

Comment:
So that one can have strings with `__json__` method. If a string does not have `__json__` method, we go by the normal flow, if it has, we use it.